packages/server/src/esignet-api.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
+ */
+
+import * as jwt from "jsonwebtoken";
+import { env } from "./constants";
+import fetch from 'node-fetch'
+import { logger } from "./logger";
+
+type OIDPUserAddress = {
+  formatted: string;
+  street_address: string;
+  locality: string;
+  region: string;
+  postal_code: string;
+  city: string;
+  country: string;
+};
+
+type OIDPUserInfo = {
+  sub: string;
+  name?: string;
+  given_name?: string;
+  family_name?: string;
+  middle_name?: string;
+  nickname?: string;
+  preferred_username?: string;
+  profile?: string;
+  picture?: string;
+  website?: string;
+  email?: string;
+  email_verified?: boolean;
+  gender?: "female" | "male";
+  birthdate?: string;
+  zoneinfo?: string;
+  locale?: string;
+  phone_number?: string;
+  phone_number_verified?: boolean;
+  address?: Partial<OIDPUserAddress>;
+  updated_at?: number;
+};
+
+const OIDP_USERINFO_ENDPOINT =
+  env.NATIONAL_ID_OIDP_REST_URL && new URL('oidc/userinfo', env.NATIONAL_ID_OIDP_REST_URL).toString()
+
+const decodeUserInfoResponse = (response: string) => {
+  return jwt.decode(response) as OIDPUserInfo;
+};
+
+export const fetchFromHearth = <T = any>(
```

Comment:
Starting to think about refactor.  We no longer think it is safe to directly query Hearth.  Instead lets use the FHIR Location API via GATEWAY_URL.  Documentation is here: https://documentation.opencrvs.org/technology/interoperability/fhir-location-rest-api